File: 03_second_largest.py
n = int(input("Enter size of array: "))
arr = list(map(int, input(f"Enter {n} elements: ").split()))
unique_arr = list(set(arr))
if len(unique_arr) < 2:
    print("No second largest element")
else:
    unique_arr.sort(reverse=True)
    print("Second largest element =", unique_arr[1])

File: 03_second_largest.c
#include <stdio.h>
#include <limits.h>

int main() {
    int n, i, first, second;
    printf("Enter size of array: ");
    scanf("%d", &n);
    int arr[n];
    printf("Enter %d elements: ", n);
    for(i=0; i<n; i++) scanf("%d", &arr[i]);
    first = second = INT_MIN;
    for(i=0; i<n; i++) {
        if(arr[i] > first) {
            second = first;
            first = arr[i];
        } else if(arr[i] > second && arr[i] < first) {
            second = arr[i];
        }
    }
    if(second == INT_MIN) printf("No second largest element\n");
    else printf("Second largest element = %d\n", second);
    return 0;
}